indico/htdocs/js/indico/Core/Dialogs/Users.js

```diff
@@ -1481,20 +1497,62 @@ type("UserListField", ["IWidget"], {
 
     draw: function() {
         var self = this;
-        var select;
-        var buttonDiv = Html.div({style:{marginTop: pixels(10)}});
+        var buttonDiv = Html.div({style: {marginTop: pixels(10)}});
 
         if (this.allowSearch || this.includeFavourites || exists(this.suggestedUsers)) {
+            var chooseUserButton = Html.input("button", {style: {marginRight: pixels(5)},
+                                                         className: 'i-button', type: 'button'},
+                                              this.enableGroups ? $T('Add User / Group'): $T('Add user'));
+            var chooseIpNetwork = Html.a({href: '#', style: {marginRight: pixels(5)},
+                                          className: 'i-button arrow js-dropdown',
+                                          'data-toggle': 'dropdown'}, $T('Add IP Network'));
+            var ipNetworksContainer = Html.span({className: 'group'});
+            var ipNetworksList = Html.ul({className: 'dropdown'});
+
+            _.each(this.ipNetworks, function(network) {
+                var li = Html.li();
+                li.append(Html.a({href: '#'}, network.name));
+                li.observeClick(function() {
+                    self.newProcess([network], function(result) {
+                        if (result && !self.userList.get(network.identifier)) {
+                            var entries = self.userList.getAll();
+                            entries[network.identifier] = $O(network);
+                            updatePrincipalsList(entries);
+                        }
+                    });
+                });
+                ipNetworksList.append(li);
+            });
 
-            var chooseUserButton = Html.input("button", {style:{marginRight: pixels(5)}, className: 'i-button'}, this.enableGroups ? $T('Add Indico User / Group'): $T('Add Indico user'));
-
-            var title = "";
+            ipNetworksContainer.append(chooseIpNetwork);
+            ipNetworksContainer.append(ipNetworksList);
+            var title;
             if (this.includeFavourites || exists(this.suggestedUsers)) {
                 title = this.enableGroups ? $T("Add Users and Groups") : $T("Add Users");
             } else {
                 title = this.enableGroups ? $T("Search Users and Groups") : $T("Search Users");
             }
 
+            function updatePrincipalsList(entries) {
+                var sortedKeys = _.sortBy(_.keys(entries), function(key) {
+                    var principal = entries[key].getAll();
+                    var weight;
+                    if (principal._type == 'Avatar') {
+                        weight = 0;
+                    } else if (principal.isGroup) {
+                        weight = 1;
+                    } else {
+                        weight = 2;
+                    }
```

Comment:
This could be a one-liner:
```javascript
var weight = (principal._type == 'Avatar') ? 0 : (principal.isGroup ? 1 : 2);
```

Not sure if too obfuscated, seems reasonably readable to me...